balrogscript/src/balrogscript/submitter/cli.py

```diff
@@ -143,24 +148,50 @@ def generate_data(self, appVersion, productName, version, buildNumber, updateCha
     def run(
         self, appVersion, productName, version, buildNumber, updateChannels, ftpServer, bouncerServer, enUSPlatforms, hashFunction, updateLine, **updateKwargs
     ):
-        data = self.generate_data(
+        blob = self.generate_data(
             appVersion, productName, version, buildNumber, updateChannels, ftpServer, bouncerServer, enUSPlatforms, updateLine, **updateKwargs
         )
         name = get_release_blob_name(productName, version, buildNumber, self.suffix)
-        api = Release(name=name, auth0_secrets=self.auth0_secrets, api_root=self.api_root)
-        try:
-            current_data, data_version = api.get_data()
-        except HTTPError as e:
-            if e.response.status_code == 404:
-                log.warning("Release blob doesn't exist, using empty data...")
-                current_data, data_version = {}, None
-            else:
+
+        if self.backend_version == 2:
+            log.info("Using backend version 2...")
+            blob["schema_version"] = self.schemaVersion
+            blob["hashFunction"] = hashFunction
+            blob["name"] = name
+            data = {"product": productName, "blob": blob}
+            headers = {"Accept-Encoding": "application/json", "Accept": "application/json", "Content-Type": "application/json", "Referer": self.api_root}
+            requests_api = get_balrog_api(auth0_secrets=self.auth0_secrets)
+            url = self.api_root + "/v2/" + name
+            logging.debug("Balrog request to {url} via PUT", url)
+            logging.debug("Data sent: %s", json.dumps(data))
+            before = time.time()
+            resp = requests_api.put(url, data=data, headers=headers)
+            try:
+                resp.raise_for_status()
+                return
+            except requests.HTTPError as excp:
+                logging.error("Caught HTTPError: %s", excp.response.content)
                 raise
+            finally:
+                stats = {"timestamp": time.time(), "method": "PUT", "url": url, "status_code": resp.status_code, "elapsed_secs": time.time() - before}
+                logging.debug("REQUEST STATS: %s", json.dumps(stats))
+            return
```

Comment:
I'm half-wanting two helper functions, but that doesn't block. Especially if we're planning to tear out support for api version 1.

Reply:
I'd support this, if for no other reason than it makes it much easier to rip out v1 support. I'm happy to defer to @Callek on how he wants to proceed though.